[SC-270] Sprawdzenie aktywności wyrzuca błąd przy obecnym configu

barto14753 opened the pull request:
@kwdrt FileTask nie miał określonego w wymaganiach daty końca przez co nie można było pobrać pierwszej odpowiedzi do tego fileTask'a. Wrzucam jako draft bo muszę jeszcze coś sprawdzić przy tym, ale po tych zmianach configu sprawdzenie aktywności nie wywala errora (przynajmniej przy pierwszym otwarciu, musze to jeszcze troche potestować)

barto14753 committed:
[SC-270] Fixed fileTask requirements

barto14753 committed:
[SC-270] Proposition for handling which fileTasks should be evaluated

barto14753 commented:
Moja propozycja zmian odnośnie oceny fileTasków

- Do tej pory w sprawdzaniu aktywności prowadzący dostawał wszystkie rozwiązania zadań, które do niego były przypisane (dlatego mógł np. ocenić zadanie, które możliwe, że jakiś student chciałby poprawić jeszcze przed końcem terminu oddania). 
-  Teraz do oceny prowadzący dostaje tylko te fileTaski, które są "zamknięte" (czyli mają w wymaganiach ustawione dataTo i to dataTo jest w danym momencie przekroczone, czyli termin oddania upłynął)

Obie opcje wydają się ok, dlatego dajcie znać co myślicie. Przez to, że prowadzący miał dostawać do oceny fileTaski, które nie miały daty zakończenia wywalało errora. Można to ominąć gdybyśmy zostali przy obecnej wersji zwracania zadań do oceny przez prowadzącego

kwdrt commented:
> Moja propozycja zmian odnośnie oceny fileTasków
> 
>     * Do tej pory w sprawdzaniu aktywności prowadzący dostawał wszystkie rozwiązania zadań, które do niego były przypisane (dlatego mógł np. ocenić zadanie, które możliwe, że jakiś student chciałby poprawić jeszcze przed końcem terminu oddania).
> 
>     * Teraz do oceny prowadzący dostaje tylko te fileTaski, które są "zamknięte" (czyli mają w wymaganiach ustawione dataTo i to dataTo jest w danym momencie przekroczone, czyli termin oddania upłynął)
> 
> 
> Obie opcje wydają się ok, dlatego dajcie znać co myślicie. Przez to, że prowadzący miał dostawać do oceny fileTaski, które nie miały daty zakończenia wywalało errora. Można to ominąć gdybyśmy zostali przy obecnej wersji zwracania zadań do oceny przez prowadzącego

Osobiście wydaje mi się że pierwsza opcja jest lepsza, z trzech powodów:
- jeśli ktoś oddał, to powinien być gotowy na to że będzie ocenione  ¯\_(ツ)_/¯
- prowadzący nie ma feedbacku, a potem jest "niedziela wieczur", termin upływa, i widzi że system się wysypał i nic nie ma? trochę słabo, lepiej widzieć w trakcie i mieć feedback od studentów
- ze strony dev-owskiej - testowanie i sprawdzanie czegokolwiek związanego z tym będzie bardzo trudne w wariancie numer 2

barto14753 committed:
[SC-270] Some refactor

barto14753 committed:
[SC-270] Merge remote-tracking branch 'origin/SC-270-sprawdzenie-aktywnosci-wyrzuca-blad-przy-obecnym-configu' into SC-270-sprawdzenie-aktywnosci-wyrzuca-blad-przy-obecnym-configu

# Conflicts:
#	api/src/main/java/com/example/api/service/activity/task/TaskService.java

barto14753 committed:
[SC-270] Took into account only selected dateTo requirements for checking if fileTaskResult is late

barto14753 commented:
Przywróciłem poprzednią wersje, czyli prowadzący może dostać do oceny fileTaski, który data ukończenia jeszcze nie nastąpiła. Teraz też nie powinno być problemu z fileTaskami, które nie mają ustalonej dateTo. DateTo jest wykorzystywane do sprawdzenia, czy fileTaskResult nie zostały wysłany z opóznieniem, jeśli dateTo nie jest określone wtedy isLate=false

kwdrt reviewed (approved):
LGTM 👍
